src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -366,12 +360,27 @@ private void WriteStreamReferenceProperty(string propertyName, ODataStreamRefere
             ODataProperty property)
         {
             this.WriterValidator.ValidateNullPropertyValue(
-                this.currentPropertyInfo.MetadataType.TypeReference, property.Name, this.Model);
+                this.currentPropertyInfo.MetadataType.TypeReference, property.Name,
+                this.currentPropertyInfo.IsTopLevel, this.Model);
 
             if (this.currentPropertyInfo.IsTopLevel)
             {
-                // TODO: Enable updating top-level properties to null #645
-                throw new ODataException("A null top-level property is not allowed to be serialized.");
+                if (this.JsonLightOutputContext.MessageWriterSettings.LibraryCompatibility <=
+                    ODataLibraryCompatibility.Version6)

```

Comment:
Should we define an ODataLibraryCompatibility.Version7, and use < ODataLibraryCompatibilityVersion7 here, in case we end up defining other 6.x compatibility versions? (which I know we don't plan to do, but might be safer to do <7 just in case?)

Reply:
I thought we discussed having a simply compatibility flag, one that would not support 6.4.2. Just 6.x, 7.x which is why the enum is a simple int matching the major version number. Did I remember wrong?

Reply:
No, you recall correctly. However, I got back to my office and started working on some of the contextUrl fixes and it occurred to me that this would be a place where I could use the library compatibility flag to make a spec-compliant change that was still a change in behavior, so I started reconsidering how we might use this in the future, and thought perhaps introducing a slightly more granular system might make sense.  I wouldn't make it a lot more complex by trying to specify support for future builds of earlier major version (maintenance) releases, but being able to distinguish, i.e., between 7.4.3 and 7.4.4. might be useful.